astropy/io/ascii/core.py

```diff
@@ -671,7 +671,7 @@ class BaseData(object):
     fill_include_names = None
     fill_exclude_names = None
     # Currently, the default matches the numpy default for masked values.
```

Comment:
Point taken about the least surprising behavior, but I think here we are getting into the distinction between the screen representation of numpy vs. a file serialization format.  I think it is pretty clear that `'--'` is no good as a unique serialization marker for missing data, whereas `''` has a fair bit of precedent.  And since we already took the much bigger  leap of making `''` indicated masked when *reading*, let's close the loop now and do the same for *writing*.